src/main/java/uk/gov/hmcts/darts/audio/helper/UnstructuredDataHelper.java

```diff
@@ -0,0 +1,146 @@
+    private final ExternalLocationTypeRepository externalLocationTypeRepository;
+    private final UserAccountRepository userAccountRepository;
+
+    private static final List<CompletableFuture> jobsList = new ArrayList<>();
+
+    public static List<CompletableFuture> getJobsList() {
+        return jobsList;
+    }
+
+    public boolean createUnstructured(ExternalObjectDirectoryEntity eodEntityToDelete, ExternalObjectDirectoryEntity eodEntity, BinaryData binaryData) {
+        boolean returnVal = true;
+        UUID uuid = saveToUnstructuredDataStore(eodEntity, binaryData);
+        if (uuid == null) {
+            returnVal = false;
+        } else {
+            saveToDatabase(eodEntity, uuid);
+            removeFromDatabase(eodEntityToDelete);
+        }
+        return returnVal;
+    }
+
+    private UUID saveToUnstructuredDataStore(ExternalObjectDirectoryEntity eodEntity, BinaryData binaryData) {
+        UUID uuid = null;
+        try {
+            Map<String, String> metadata = new HashMap<>();
+            metadata.put(MEDIA_REQUEST_ID, String.valueOf(eodEntity.getMedia().getId()));
+            BlobClient blobClient = dataManagementApi.saveBlobDataToContainer(binaryData, DatastoreContainerType.UNSTRUCTURED, metadata);
+            uuid = UUID.fromString(blobClient.getBlobName());
+            log.debug(
+                "Completed upload to unstructured data store for mediaRequestId {}. Successfully uploaded with blobId: {}",
+                eodEntity.getMedia().getId(),
+                uuid
+            );
+        } catch (Exception e) {
+            log.error(
+                "Upload to unstructured data store failed for mediaRequestId {}.",
+                eodEntity.getMedia().getId()
+            );
+        }
+        return uuid;
+    }
+
+    @Transactional
+    private void saveToDatabase(ExternalObjectDirectoryEntity eod, UUID uuid) {
+        ExternalObjectDirectoryEntity eodUnstructured = new ExternalObjectDirectoryEntity();
+
+        MediaEntity mediaEntity = eod.getMedia();
+        if (mediaEntity != null) {
+            eodUnstructured.setMedia(mediaEntity);
+        }
+        TranscriptionDocumentEntity transcriptionDocumentEntity = eod.getTranscriptionDocumentEntity();
+        if (transcriptionDocumentEntity != null) {
+            eodUnstructured.setTranscriptionDocumentEntity(transcriptionDocumentEntity);
+        }
+        AnnotationDocumentEntity annotationDocumentEntity = eod.getAnnotationDocumentEntity();
+        if (annotationDocumentEntity != null) {
+            eodUnstructured.setAnnotationDocumentEntity(annotationDocumentEntity);
+        }
+        CaseDocumentEntity caseDocumentEntity = eod.getCaseDocument();
+        if (caseDocumentEntity != null) {
+            eodUnstructured.setCaseDocument(caseDocumentEntity);
+        }
+        eodUnstructured.setExternalLocation(uuid);
+        OffsetDateTime now = OffsetDateTime.now();
+        eodUnstructured.setCreatedDateTime(now);
```

Comment:
The ExternalObjectDirectoryEntity extends CreatedModifiedBaseEntity which has annotations on these 2 fields to stamp the time, so they are a bit redundant?

Reply:
removed